fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/mixin/registry/sync/MixinMain.java

```diff
@@ -0,0 +1,40 @@
+package net.fabricmc.fabric.mixin.registry.sync;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.concurrent.CompletableFuture;
+
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import com.mojang.authlib.GameProfileRepository;
+import com.mojang.authlib.minecraft.MinecraftSessionService;
+import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.resource.DataPackSettings;
+import net.minecraft.resource.ResourcePackManager;
+import net.minecraft.resource.ServerResourceManager;
+import net.minecraft.server.Main;
+import net.minecraft.server.dedicated.EulaReader;
+import net.minecraft.server.dedicated.ServerPropertiesLoader;
+import net.minecraft.util.UserCache;
+import net.minecraft.util.dynamic.RegistryOps;
+import net.minecraft.util.registry.DynamicRegistryManager;
+import net.minecraft.world.level.storage.LevelStorage;
+
+import net.fabricmc.fabric.impl.registry.sync.PersistentDynamicRegistryHandler;
+
+@Mixin(Main.class)
+public class MixinMain {
+	@SuppressWarnings("rawtypes")
+	@Inject(method = "main", at = @At(value = "INVOKE_ASSIGN", target = "Lnet/minecraft/util/dynamic/RegistryOps;of(Lcom/mojang/serialization/DynamicOps;Lnet/minecraft/resource/ResourceManager;Lnet/minecraft/util/registry/DynamicRegistryManager$Impl;)Lnet/minecraft/util/dynamic/RegistryOps;"), locals = LocalCapture.CAPTURE_FAILHARD)
+	private static void main(String[] args, CallbackInfo ci, OptionParser optionParser, OptionSpec optionSpec, OptionSpec optionSpec2, OptionSpec optionSpec3, OptionSpec optionSpec4, OptionSpec optionSpec5, OptionSpec optionSpec6, OptionSpec optionSpec7, OptionSpec optionSpec8, OptionSpec optionSpec9, OptionSpec optionSpec10, OptionSpec optionSpec11, OptionSpec optionSpec12, OptionSpec optionSpec13, OptionSpec optionSpec14, OptionSet optionSet, DynamicRegistryManager.Impl impl, Path path, ServerPropertiesLoader serverPropertiesLoader, Path path2, EulaReader eulaReader, File file, YggdrasilAuthenticationService yggdrasilAuthenticationService, MinecraftSessionService minecraftSessionService, GameProfileRepository gameProfileRepository, UserCache userCache, String string, LevelStorage levelStorage, LevelStorage.Session session, DataPackSettings dataPackSettings, boolean bl, ResourcePackManager resourcePackManager, DataPackSettings dataPackSettings2, CompletableFuture completableFuture, ServerResourceManager serverResourceManager2, RegistryOps registryOps) {
```

Comment:
Just a few locals ;)

Reply:
o p t i o n s p e c

Reply:
It might not be a bad idea to look at using a redirect here, or a ModifyArg as this mixin is going to be super fragile to updates.

Reply:
Yeah I had a similar thing in the server world load event. The local variable are insanity. If you do choose that please note with a comment about how the redirect is easier to maintain than a modify arg due to large amount of local variables.

Reply:
I have moved to 2 redirects.